src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { FilterType } from '../../types/FilterType';
+
+type FooterProps = {
+  todos: Todo[];
+  filter: FilterType;
+  onFilterChange: (filter: FilterType) => void;
+  onClearCompleted: () => void; // Add onClearCompleted prop
+};
+
+export const Footer: React.FC<FooterProps> = ({
+  todos,
+  filter,
+  onFilterChange,
+  onClearCompleted, // Destructure onClearCompleted prop
+}) => {
+  if (todos.length === 0) {
+    return null;
+  }
+
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedTodosCount = todos.filter(todo => todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {activeTodosCount} items left
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={`filter__link ${filter === FilterType.All ? 'selected' : ''}`}
+          data-cy="FilterLinkAll"
+          onClick={() => onFilterChange(FilterType.All)}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={`filter__link ${filter === FilterType.Active ? 'selected' : ''}`}
+          data-cy="FilterLinkActive"
+          onClick={() => onFilterChange(FilterType.Active)}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={`filter__link ${filter === FilterType.Completed ? 'selected' : ''}`}
+          data-cy="FilterLinkCompleted"
+          onClick={() => onFilterChange(FilterType.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      <button
+        type="button"
+        className="todoapp__clear-completed"
+        data-cy="ClearCompletedButton"
+        disabled={completedTodosCount === 0}
+        onClick={onClearCompleted} // Add onClick handler
+      >
+        Clear completed
```

Comment:
There is clearly the same pattern for every anchor. What about creating array with objects that keep the values needed to properly create every anchor with map.